xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/redirection/RedirectClassRedirectionFilter.java

```diff
@@ -0,0 +1,102 @@
+ */
+package com.xpn.xwiki.internal.redirection;
+
+import java.io.IOException;
+
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.resource.ResourceReference;
+import org.xwiki.resource.ResourceReferenceManager;
+import org.xwiki.resource.entity.EntityResourceReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.internal.mandatory.RedirectClassDocumentInitializer;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.redirection.RedirectionFilter;
+import com.xpn.xwiki.web.Utils;
+
+/**
+ * Perform a redirection based on the presence of a {@code XWiki.RedirectClass} XClass.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Singleton
+public class RedirectClassRedirectionFilter implements RedirectionFilter
+{
+    @Override
+    public boolean redirect(XWikiContext context) throws XWikiException
+    {
+        WikiReference wikiReference = context.getWikiReference();
+
+        // Look if the document has a redirect object
+        XWikiDocument doc = context.getDoc();
+        BaseObject redirectObj = doc.getXObject(RedirectClassDocumentInitializer.REFERENCE);
+        if (redirectObj == null) {
+            return false;
+        }
+
+        // Get the location
+        String location = redirectObj.getStringValue("location");
+        if (StringUtils.isBlank(location)) {
+            return false;
+        }
+
+        // Resolve the location to get a reference
+        DocumentReferenceResolver<String> resolver = Utils.getComponent(DocumentReferenceResolver.TYPE_STRING);
+        EntityReference locationReference = resolver.resolve(location, wikiReference);
+
+        // Get the type of the current target
+        ResourceReference resourceReference = Utils.getComponent(ResourceReferenceManager.class).getResourceReference();
+        EntityResourceReference entityResource = (EntityResourceReference) resourceReference;
+        EntityReference entityReference = entityResource.getEntityReference();
+
+        // If the entity is inside a document, compute the new entity with the new document part.
+        if (entityReference.getType().ordinal() > EntityType.DOCUMENT.ordinal()) {
+            EntityReference parentDocument = entityReference.extractReference(EntityType.DOCUMENT);
+            locationReference = entityReference.replaceParent(parentDocument, locationReference);
+        }
+
+        // Get the URL corresponding to the location
+        // Note: the anchor part is lost in the process, because it is not sent to the server
+        // (see: http://stackoverflow.com/a/4276491)
```

Comment:
I know it's a move, but I don't see the point of putting a stackoverflow post as reference... The explanation in the line above is enough.

Reply:
Indeed, I haven't checked the moved code closely. I'll remove it.